src/test/unit_tests/jest_tests/test_nc_lifecycle_posix_integration.test.js

```diff
@@ -558,36 +589,336 @@ describe('noobaa nc - lifecycle versioning ENABLE', () => {
+            await object_sdk.delete_object({ bucket: test_bucket, key: test_key1_regular });
+            await object_sdk.delete_object({ bucket: test_bucket, key: test_key2_regular }); //last in batch should not delete
+            await object_sdk.delete_object({ bucket: test_bucket, key: test_key2_regular });
+            await object_sdk.delete_object({ bucket: test_bucket, key: test_key3_regular }); // last in batch should delete
+            await object_sdk.delete_object({ bucket: test_bucket, key: 'test_key4' });
 
             await exec_manage_cli(TYPES.LIFECYCLE, '', { disable_service_validation: 'true', disable_runtime_validation: 'true', config_root }, undefined, undefined);
             const object_list = await object_sdk.list_object_versions({ bucket: test_bucket });
-            expect(object_list.objects.length).toBe(4); //added delete marker
-            let has_delete_marker = false;
+            expect(object_list.objects.length).toBe(4);
             object_list.objects.forEach(element => {
-                if (element.delete_marker) has_delete_marker = true;
+                expect([test_key1_regular, test_key2_regular]).toContain(element.key);
             });
-            expect(has_delete_marker).toBe(true);
         });
+    });
+
+});
+
+describe('noobaa nc - lifecycle versioning SUSPENDED', () => {
+    const test_bucket_path = `${root_path}/${test_bucket}`;
+    const test_key1_regular = 'test_key1';
+    const test_key2_regular = 'test_key2';
+    const test_key3_regular = 'test_key3';
+    const test_key1_nested = 'nested/test_key1';
+    const test_key2_nested = 'nested/test_key2';
+    const test_key3_nested = 'nested/test_key3';
+    const prefix = 'test/';
+    const test_prefix_key_regular = `${prefix}${test_key1_regular}`;
+    const test_prefix_key_nested = `${prefix}${test_key1_nested}`;
+    const test_cases = [
+        { description: 'regular key', test_key1: test_key1_regular, test_key2: test_key2_regular, test_key3: test_key3_regular, test_prefix_key: test_prefix_key_regular },
+        { description: 'nested key', test_key1: test_key1_nested, test_key2: test_key2_nested, test_key3: test_key3_nested, test_prefix_key: test_prefix_key_nested },
+    ];
+    let object_sdk;
+
+    beforeAll(async () => {
+        await fs_utils.create_fresh_path(config_root, 0o777);
+        set_nc_config_dir_in_config(config_root);
+        await fs_utils.create_fresh_path(root_path, 0o777);
+        const res = await exec_manage_cli(TYPES.ACCOUNT, ACTIONS.ADD, account_options1);
+        const json_account = JSON.parse(res).response.reply;
+        console.log(json_account);
+        object_sdk = new NsfsObjectSDK('', config_fs, json_account, "DISABLED", config_fs.config_root, undefined);
+        object_sdk.requesting_account = json_account;
+        await object_sdk.create_bucket({ name: test_bucket });
+        await object_sdk.set_bucket_versioning({ name: test_bucket, versioning: "SUSPENDED" });
```

Comment:
since you set it to be Suspended from the beginning. wouldn't it mean we will never have more then one version per object?

Reply:
it is good to have a single test that we test that we actually don't do anything

Reply:
There are tests where the bucket versioning was set to SUSPENDED at the beginning, for example:
- `nc lifecycle - versioning SUSPENDED - expiration rule - regular key`
- `nc lifecycle - versioning SUSPENDED - noncurrent expiration rule - expire older versions - `
in the rest of the tests, the bucket versioning is set to ENABLED and then to SUSPENDED, so they are just another validation of the original case.